app/Controllers/TimeSlot.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace App\Controllers;
+
+use App\Helpers\TimeSlotData;
+use App\Models\EventModel;
+use App\Models\TimeSlotModel;
+use CodeIgniter\HTTP\ResponseInterface;
+
+class TimeSlot extends BaseController
+{
+    const TIME_SLOT_RULES = [
+        'time_slots.*.start_time' => 'required|valid_date[Y-m-d H:i:s]',
+        'time_slots.*.duration' => 'required|is_natural_no_zero',
+    ];
+
+    public function create_or_replace(int $eventId): ResponseInterface
+    {
+        $data = $this->request->getJSON(true);
+
+        if (!$this->validateData($data, self::TIME_SLOT_RULES)) {
+            return $this->response->setJSON($this->validator->getErrors())->setStatusCode(400);
+        }
+
+        $validData = $this->validator->getValidated();
+
+        $eventModel = model(EventModel::class);
+        $event = $eventModel->get($eventId);
+        if ($event === null) {
+            return $this->response->setJSON(['error' => 'EVENT_NOT_FOUND'])->setStatusCode(404);
+        }
+
+        $timeSlotModel = model(TimeSlotModel::class);
+
+        $timeSlots = array_map(
+            function (array $slot) use ($eventId) {
+                return TimeSlotData::make(
+                    $eventId,
+                    $slot['start_time'],
+                    $slot['duration'],
+                );
+            },
+            $validData['time_slots'],
+        );
+
+        $validationResult = $this->areTimeSlotsValid($timeSlots, $event);
+        if ($validationResult !== true) {
+            return $validationResult;
+        }
+
+        $timeSlotModel->deleteAllOfEvent($eventId);
```

Comment:
Does this mean, that once you added Speaker to a Slot you never be able to update the slots then?
Because the ID changes?
Is that a problem?

Reply:
That's correct, yes. In my opinion, we should not change the slots after we started assigning talks to them, because this would make everything really chaotic. If we really need to delete/change/create *single* slots, we can add that functionality later. But I dearly hope that we won't need that feature.
Is that okay for you?

Reply:
We did change the slots last time. Didn't we?
We did not had the 30 minutes talk in the beginning.
Or am I wrong?

Reply:
I don't remember exactly. But we can delay the creation of the time slots until after the end of the call for papers time frame. I think that should be enough.
However, in the worst case, we can still edit them manually (or add additional logic and a dashboard menu to edit them). In general, I would just not recommend doing this, because I think it could create chaos.

Reply:
okay